product_variant_default_code/views/product_view.xml

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<odoo>
+
+        <record id="product_template_only_form_view_inh_variantdefaultcode"
+                model="ir.ui.view">
+            <field name="name">
+                product.template.only.form.view.inh.variantdefaultcode
+            </field>
+            <field name="model">product.template</field>
+            <field name="inherit_id"
+                   ref="product.product_template_only_form_view"/>
+            <field name="arch" type="xml">
+                <field name="default_code" position="before">
+                  <field name="reference_mask"
+                         modifiers="{'invisible':
```

Comment:
@chienandalu @pedrobaeza Are you sure this line works.  I had it in my own code and it didn't.  It needed

attrs={....[('attribute_line_ids', '=', [])]}

something like that anyway, attrs not modifiers, empty list not False.